src/main/aerospike.c

```diff
@@ -83,187 +88,535 @@ static PyMethodDef Aerospike_Methods[] = {
+            goto GLOBAL_HOSTS_CLEANUP_ON_ERROR;
+        }
     }
-    Aerospike_State(aerospike)->kdict = kdict;
 
     /*
 	 * Add constants to module.
 	 */
-    for (i = 0; i < (int)OPERATOR_CONSTANTS_ARR_SIZE; i++) {
-        PyModule_AddIntConstant(aerospike, operator_constants[i].constant_str,
-                                operator_constants[i].constantno);
-    }
-
-    for (i = 0; i < (int)AUTH_MODE_CONSTANTS_ARR_SIZE; i++) {
-        PyModule_AddIntConstant(aerospike, auth_mode_constants[i].constant_str,
-                                auth_mode_constants[i].constantno);
-    }
-
-    declare_policy_constants(aerospike);
-    declare_log_constants(aerospike);
-
-    PyObject *predicates = AerospikePredicates_New();
-    Py_INCREF(predicates);
-    retval = PyModule_AddObject(aerospike, "predicates", predicates);
-    if (retval == -1) {
-        goto CLEANUP;
-    }
-    Aerospike_State(aerospike)->predicates = predicates;
-
-    PyTypeObject *geospatial = AerospikeGeospatial_Ready();
-    Py_INCREF(geospatial);
-    retval = PyModule_AddObject(aerospike, "GeoJSON", (PyObject *)geospatial);
-    if (retval == -1) {
-        goto CLEANUP;
-    }
-    Aerospike_State(aerospike)->geospatial = geospatial;
-
-    PyTypeObject *null_object = AerospikeNullObject_Ready();
-    Py_INCREF(null_object);
-    retval = PyModule_AddObject(aerospike, "null", (PyObject *)null_object);
-    if (retval == -1) {
-        goto CLEANUP;
-    }
-    Aerospike_State(aerospike)->null_object = null_object;
-
-    PyTypeObject *wildcard_object = AerospikeWildcardObject_Ready();
-    Py_INCREF(wildcard_object);
-    retval = PyModule_AddObject(aerospike, "CDTWildcard",
-                                (PyObject *)wildcard_object);
-    if (retval == -1) {
-        goto CLEANUP;
-    }
-    Aerospike_State(aerospike)->wildcard_object = wildcard_object;
-
-    PyTypeObject *infinite_object = AerospikeInfiniteObject_Ready();
-    Py_INCREF(infinite_object);
-    retval = PyModule_AddObject(aerospike, "CDTInfinite",
-                                (PyObject *)infinite_object);
-    if (retval == -1) {
-        goto CLEANUP;
+    for (i = 0; i < sizeof(module_constants) / sizeof(module_constants[0]);
+         i++) {
+        if (module_constants[i].is_str_value == false) {
+            retval = PyModule_AddIntConstant(py_aerospike_module,
+                                             module_constants[i].name,
+                                             module_constants[i].value.integer);
+        }
+        else {
+            retval = PyModule_AddStringConstant(
+                py_aerospike_module, module_constants[i].name,
+                module_constants[i].value.string);
+        }
+
+        if (retval == -1) {
+            goto GLOBAL_HOSTS_CLEANUP_ON_ERROR;
+        }
     }
-    Aerospike_State(aerospike)->infinite_object = infinite_object;
 
-    return aerospike;
+    return py_aerospike_module;
 
-CLEANUP:
-    Aerospike_Clear(aerospike);
```

Comment:
Double checking that Aerospike_clear() isn't needed?

Reply:
I don't think it's needed because I'm removing the aerospike module's "state" in this PR. The objects in the module state are already added to the module. And I looked up C code containing ".m_clear" on Github and most people use `m_clear` only to clear the state of a module. 

Ram added `Aerospike_Clear()` while working on the old async features, but he didn't explain why he added it: https://github.com/aerospike/aerospike-client-python/commit/84eaaa41101ba379eb78dd5f420dcbac5271ad00